cltk/text_reuse/automata.py

```diff
@@ -0,0 +1,452 @@
+
+            >>> A = DFA({'q1', 'q2'}, ['0', '1'], 'q1', set())
+
+        Adding final states:
+            States can be added to F even after initializing
+
+            >>> A.add_final_state('q1')
+
+        Adding transitions:
+            You can either define the transition table by assigning
+            a nested dic (see further down for example) to the delta
+            parameter at initialization, or manually define each
+            transition
+
+            >>> A.add_transition('q1', '0', 'q2')
+
+            >>> A.add_transition('q2', '1', 'q2')
+
+            >>> A.add_transition('q2', '0', 'q1')
+
+            The equivalent transition trie will be this:
+
+            >>> A.transition
+            {'q1': {'0': 'q2'}, 'q2': {'1': 'q2', '0': 'q1'}}
+
+        Calling the transition function:
+            To call δ(qi, u), simply call transition_function:
+
+            >>> A.transition_function('q1', '0')
+            'q2'
+
+            The method returns null if the transition is not defined
+
+            >>> A.transition_function('q1', '1')
+            
+
+        Accepted input strings:
+            Determining whether an input belongs to the language
+            recognized by a DFA A only is O(|w|) time, in contrast
+            to the exponential solution of the equivalnet problem
+            using a nondeterministic automaton.
+
+            Since the automaton is simple, we can deduce its language
+            without the need for any calculations: (01*0)*
+
+            Now on to testing
+
+            >>> A.accepted('01000')
+            True
+
+            >>> A.accepted('010011001110')
+            True
+
+            >>> A.accepted('100100')
+            False
+    """
+
+    def __init__(self, Q, S, s, F, delta = False):
+        """
+        :param Q: set: Set of states Q: q0, q1, ... qn
+        :param S: str set: Set of symbols recognized by the automaton
+        :param s: int: The starting state q_s1
+        :param F: int set: Set of accepted states {q'0, ... q'|F|)
+        :param delta: Nq*|S| int transition matrix (defaults to empty transition matrix)
+        """
+
+        self.Q = Q
+        self.S = S
+
+        if s not in self.Q:
+            print("The specified value is invalid, s must be a member of Q")
+            raise ValueError
+        else:
+            self.s = s
+
+        if sum([f not in self.Q for f in F]) != 0:
+            print("The specified value is invalid, F must be a subset of Q (F∩Q≠∅)")
+            raise ValueError
+        else:
+            self.F = F
+
+        if delta:
+            self.transition = delta
+        else:
+            self.transition = {}
+
+    def add_final_state(self, f):
+
+        """
+        :param f: int , the state qi to be added to F, epsilon is
+        conventionally defined as the last node (q_|S|)
+        """
+        if f not in  self.Q:
+            print("The specified value is invalid, f must be a member of Q")
```

Comment:
change to LOG.info(...)
raw print statements are considered undesirable in a library